src/energy.h

```diff
@@ -0,0 +1,157 @@
+#ifndef ENERGY_H
+#define ENERGY_H
+
+#include <utility>
+
+#include "rng.h"
+
+class JsonIn;
+class JsonOut;
+
+class energy_quantity
```

Comment:
What's the advantage of this class over the existing `units::quantity` template class?

Reply:
None really, I just hadn't considered using `units::quantity`.